os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -489,7 +481,8 @@ void up_assert(const uint8_t *filename, int lineno)
 		lldbg("No heap corruption detected\n");
 	}
 #ifdef CONFIG_APP_BINARY_SEPARATION
-	if (!is_kernel_fault) {
+	if (fault_tcb->uheap != NULL) {
```

Comment:
Is it possible to redefine belows for readability?
1. ault_tcb->uheap != NULL  <=  fault thread
2. is_kernel_space((void *)assert_pc   <= fault space